content/zh/blog/_posts/2022-02-17-updated-dockershim-faq.md

```diff
@@ -0,0 +1,386 @@
+
+<!--
+### What should I look out for when changing CRI implementations?
+-->
+### 当切换 CRI 实现时，应该注意什么？
+
+<!--
+While the underlying containerization code is the same between Docker and most
+CRIs (including containerd), there are a few differences around the edges. Some
+common things to consider when migrating are:
+-->
+虽然 Docker 和大多数 CRI（包括 containerd）之间的底层容器化代码是相同的，
+但其周边部分却存在差异。迁移时要考虑如下常见事项：
+
+<!--
+- Logging configuration
+- Runtime resource limitations
+- Node provisioning scripts that call docker or use docker via it's control socket
+- Kubectl plugins that require docker CLI or the control socket
+- Tools from the Kubernetes project that require direct access to Docker Engine
+  (for example: the deprecated `kube-imagepuller` tool)
+- Configuration of functionality like `registry-mirrors` and insecure registries 
+- Other support scripts or daemons that expect Docker Engine to be available and are run
+  outside of Kubernetes (for example, monitoring or security agents)
+- GPUs or special hardware and how they integrate with your runtime and Kubernetes
+-->
+- 日志配置
+- 运行时的资源限制
+- 调用 docker 或通过其控制套接字使用 docker 的节点配置脚本
+- 需要访问 docker 命令或控制套接字的 kubectl 插件
+- 需要直接访问 Docker Engine 的 Kubernetes 工具（例如:已弃用的' kube-imagepuller '工具）
+- `registry-mirrors` 和不安全注册表等功能的配置
+- 保障 Docker Engine 可用、且运行在 Kubernetes 之外的脚本或守护进程（例如：监视或安全代理）
+- GPU 或特殊硬件，以及它们如何与你的运行时和 Kubernetes 集成
+
+<!--
+If you use Kubernetes resource requests/limits or file-based log collection
+DaemonSets then they will continue to work the same, but if you’ve customized
+your `dockerd` configuration, you’ll need to adapt that for your new container
+runtime where possible.
+-->
+如果你只是用了 Kubernetes 资源请求/限制或基于文件的日志收集 DaemonSet，它们将继续稳定工作，
+但是如果你用了自定义了 dockerd 配置，则可能需要为新的容器运行时做一些适配工作。
+
+<!--
+Another thing to look out for is anything expecting to run for system maintenance
+or nested inside a container when building images will no longer work. For the
+former, you can use the [`crictl`][cr] tool as a drop-in replacement (see [mapping from docker cli to crictl](https://kubernetes.io/docs/tasks/debug-application-cluster/crictl/#mapping-from-docker-cli-to-crictl)) and for the
+latter you can use newer container build options like [img], [buildah],
+[kaniko], or [buildkit-cli-for-kubectl] that don’t require Docker.
+-->
+另外还有一个需要关注的点，那就是当创建镜像时，系统维护或嵌入容器方面的任务将无法工作。
+对于前者，可以用 [`crictl`][cr]工具作为临时替代方案
+(参阅 [从 docker cli 到 crictl 的映射](https://kubernetes.io/zh/docs/tasks/debug-application-cluster/crictl/#mapping-from-docker-cli-to-crictl)) 。
+对于后者，可以用新的容器创建选项，例如 [img]、[buildah]、[kaniko] 或 [buildkit-cli-for-kubectl]，
+他们都不需要 Docker。
+
+[cr]: https://github.com/kubernetes-sigs/cri-tools
+[img]: https://github.com/genuinetools/img
+[buildah]: https://github.com/containers/buildah
+[kaniko]: https://github.com/GoogleContainerTools/kaniko
+[buildkit-cli-for-kubectl]: https://github.com/vmware-tanzu/buildkit-cli-for-kubectl
```

Comment:
[buildkit-cli-for-kubectl]: https://github.com/vmware-tanzu/buildkit-cli-for-kubectl。